src/lib.rs

```diff
@@ -164,4 +165,42 @@ mod tests {
 
         assert!(result.len() == 6);
     }
+
+    #[test]
+    fn merge_tester() {
```

Comment:
Can we give this a nicer name now that it's staying

```suggestion
    fn merging_structs_with_hash_implementations_deduplicates_correctly {
```